libsolidity/formal/SMTChecker.cpp

```diff
@@ -695,6 +782,18 @@ smt::Expression SMTChecker::maxValue(IntegerType const& _t)
 smt::Expression SMTChecker::expr(Expression const& _e)
 {
 	if (!m_expressions.count(&_e))
+	{
+		m_errorReporter.warning(_e.location(), "Internal error: Expression undefined for SMT solver." );
```

Comment:
Shouldnt these be assertions?

Reply:
If we leave them as warnings, you can still use the SMT checker on some functions while element of other functions are not yet supported.

Reply:
What kind of internal error still ensures that the rest is unaffected? (Not sure when the above can happen.)

Reply:
Ah sorry, this is only due to a missing implementation for an AST node type.